src/dwn.ts

```diff
@@ -1,9 +1,11 @@
 import type { BaseMessage } from './core/types.js';
 import type { DidMethodResolver } from './did/did-resolver.js';
 import type { MessageStore } from './store/message-store.js';
+import type { MethodHandler } from './interfaces/types.js';
 import type { Readable } from 'readable-stream';
-import type { Interface, MethodHandler } from './interfaces/types.js';
+import type { TenantGate } from './core/tenant-gate.js';
 
+import { AllowAllTenantGate } from './core/tenant-gate.js';
```

Comment:
Curious why this doesn't have `import type` ?

Reply:
@andresuribe87, because it is not a "type" declaration (e.g. such as an interface). It is an actual implementation (class). If no member of `AllowAllTenantGate` was ever accessed in this file, then I could add the `type` keyword as a best practice to help treeshaking by JS bundlers.